src/PostgREST/Error.hs

```diff
@@ -214,6 +216,51 @@ instance JSON.ToJSON ApiRequestError where
     "details" .= JSON.Null,
     "hint"    .= ("Try renaming the parameters or the function itself in the database so function overloading can be resolved" :: Text)]
 
+-- |
+-- If no relationship is found between a parent and a child, then it looks for the parent first.
+-- If the parent is not found then it does a fuzzy search to all the parents in the schema cache and
+-- gives the best match as suggestion. Otherwise, it does a fuzzy search to all the corresponding children
+-- of that parent and gives the best match as suggestion. If both are found, then no suggestion is given.
+--
+-- >>> :set -Wno-missing-fields
+-- >>> let qi t = QualifiedIdentifier "api" t
+-- >>> let rel ft = Relationship{relForeignTable = qi ft}
+-- >>> let rels = HM.fromList [((qi "films", "api"), [rel "directors", rel "roles", rel "actors"])]
+--
+-- >>> noRelBetweenHint "film" "directors" "api" rels
+-- Just "Perhaps you meant 'films' instead of 'film'."
+--
+-- >>> noRelBetweenHint "films" "role" "api" rels
+-- Just "Perhaps you meant 'roles' instead of 'role'."
+--
+-- >>> noRelBetweenHint "films" "role" "api" rels
+-- Just "Perhaps you meant 'roles' instead of 'role'."
+--
+-- >>> noRelBetweenHint "films" "actors" "api" rels
+-- Nothing
+--
+-- >>> noRelBetweenHint "noclosealternative" "roles" "api" rels
+-- Nothing
+--
+-- >>> noRelBetweenHint "films" "noclosealternative" "api" rels
+-- Nothing
+--
+-- >>> noRelBetweenHint "films" "noclosealternative" "noclosealternative" rels
+-- Nothing
+--
+noRelBetweenHint :: Text -> Text -> Text -> RelationshipsMap -> Maybe Text
+noRelBetweenHint parent child schema allRels = ("Perhaps you meant '" <>) <$>
+  if isJust findParent
+    then (<> "' instead of '" <> child <> "'.") <$> suggestChild
+    else (<> "' instead of '" <> parent <> "'.") <$> suggestParent
+  where
+    findParent = HM.lookup (QualifiedIdentifier schema parent, schema) allRels
+    fuzzySetOfParents  = Fuzzy.fromList [qiName (fst p) | p <- HM.keys allRels, snd p == schema]
```

Comment:
I wonder if this is expensive if `allRels` is big, in which case perhaps the hint could be disabled by looking at the map length?

Maybe this could be tested with the apflora schema, by doing `time curl ..` on a request with bad names.

Reply:
It is much slower than when a relationship is found, not sure if it's too slow, though. For instance these are the times that I got:

1. Relationship is found:
    ```bash
    $ curl -w "%{time_total}s" -o /dev/null -s "http://localhost:3000/ziel?select=ap(*)"
    0.002485s
    0.002125s
    ```

2. Relationship not found (parent not found)
    ```bash
    $ curl -w "%{time_total}s" -o /dev/null -s "http://localhost:3000/ziels?select=ap(*)"
    0.072776s
    0.056077s
    ``` 

3. Relationship not found (child not found)
    ```bash
    $ curl -w "%{time_total}s" -o /dev/null -s "http://localhost:3000/ziel?select=app(*)"
    0.035553s
    0.027014s
    ```

Reply:
> (fuzzy search for parent)
> 0.072776s
> 0.056077s
> (fuzzy search for child)
> 0.035553s
>0.027014s

So parent takes longer than child because it searches on the full `allRels`, while child only searches on the relationships that have parent. Makes sense.